### PR TITLE
Crud

### PR DESCRIPTION
**Pull request by msyurikim**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+## CRUD
+<!-- Todo: API calls for POST and DELETE -->
+<!-- PATCH instead of PUT -->
+
+| Http Verbs | Endpoint     | Action                                 | Error ?            |
+|------------|--------------|----------------------------------------|--------------------|
+| POST       | /createGame  | Creates a game with announcements      |                    |
+| GET        | /getGame     | Gets a list of reviews for 1 game      | Error: Bad Request |
+| GET        | /randomGame  | Gets all games, then picks 1 at random |                    |
+| PATCH      | /updateLikes | Updates a review for 1 game            | Error: Bad Request |
```
Comment (ingridclaire) [inline README.md]: do you know the difference between patch/put? I used put, so just wondering..

File: database/index.js [hunk 2]
```diff
@@ -113,4 +113,19 @@ module.exports = {
       },
     );
   },
+  patchGame: ({
+    id, name,
+  }, callback) => {
+    Game.findOneAndUpdate(
+      { 'gameNumber': id },
+      { $set: { 'name' : name} },
```
[the remark this answers is not in the document]
Comment (msyurikim) [reply database/index.js:121]: It's the syntax for findOneAndUpdate mongo method. The name is an attribute of the Game model (not nested).

File: index.js [hunk 3]
```diff
@@ -61,10 +65,54 @@ app.patch('/updateLikes', (req, res) => {
...
+    } else {
+      res.send(`${data.gameNumber} has been posted`);
+    }
+  });
+});
+
+app.get('/game/:id', (req, res) => {
+  db.getGame({ _id: req.params.id}, (err, data) => {
+    if (err) {
+      res.send(400);
```
[the remark this answers is not in the document]
Comment (msyurikim) [reply index.js:83]: It works, but I think it's deprecated. So you're right, it's better to do res.status(400).end(); I'll change that

File: index.js [hunk 4]
```diff
@@ -61,10 +65,54 @@ app.patch('/updateLikes', (req, res) => {
...
+    } else {
+      res.send(data);
```
[the remark this answers is not in the document]
Comment (msyurikim) [reply index.js:85]: I don't think it's necessary if you're sending back data. I don't see it being used unless you're not sending back any data in the Express docs, but I may be wrong.

File: index.js [hunk 5]
```diff
@@ -61,10 +65,54 @@ app.patch('/updateLikes', (req, res) => {
...
+      });
+    }
+  });
+});
 
-// app.listen(port, () => {
-//   console.log(`Serving is now listening on port: ${port}`);
-// });
+app.delete('/game/:id', (req, res) => {
+  db.Game.remove({gameNumber: req.params.id}, (err, data) => {
```
[the remark this answers is not in the document]
Comment (msyurikim) [reply index.js:109]: I didn't want to set up too many controllers and change his files as much, but I may change that later if this is bad practice. Yes, this is a direct db query.

File: README.md (hunk 1 above)
Comment (msyurikim) [reply README.md]: So patch updates parts of the object, but put replaces the object.

**Pull request merged**